R/object_usage_linter.R

```diff
@@ -87,8 +87,6 @@ object_usage_linter <- function(interpret_glue = TRUE, skip_with = TRUE) {
         skip_with = skip_with
       )
 
-      # TODO handle assignment functions properly
```

Comment:
I don't know what this TODO is talking about, exactly. I tried these snippets and they lint as expected:

```r
lint(linters = object_usage_linter(), "
foo <- function() {
  not_exists(x) <- 2
  not_exists(x)
}")
# <text>:3:3: warning: [object_usage_linter] no visible global function definition for 'not_exists<-'
#   not_exists(x) <- 2
#   ^~~~~~~~~~
# <text>:4:3: warning: [object_usage_linter] no visible global function definition for 'not_exists'
#   not_exists(x)
#   ^~~~~~~~~~
lint(linters = object_usage_linter(), "
not_exists <- function(x) x
foo <- function() {
  not_exists(x) <- 2
  not_exists(x)
}")
# <text>:4:3: warning: [object_usage_linter] no visible global function definition for 'not_exists<-'
#   not_exists(x) <- 2
#   ^~~~~~~~~~
```

So deleting and waiting for something to be reported in tracker if there's actually something there.

Reply:
Do we have a test for these?

Reply:
Not sure... #2451

Reply:
#2451